### PR TITLE
Require OR REPLACE for CREATE SNAPSHOT

### PR DESCRIPTION
**Pull request by viliam-durina**

### CODE REVIEW

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/SqlCreateSnapshot.java [hunk 1]
```diff
@@ -45,10 +46,15 @@
     public SqlCreateSnapshot(
             SqlIdentifier snapshotName,
             SqlIdentifier jobName,
+            boolean replace,
             SqlParserPos pos
     ) {
         super(OPERATOR, pos, true, false);
 
+        if (!replace) {
+            throw QueryException.error("The OR REPLACE option is required for CREATE SNAPSHOT");
```
Comment (gierlachg) [inline hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/SqlCreateSnapshot.java:55]: Throw `SqlValidatorException` (see `ParserResource`)?

Comment (viliam-durina) [reply hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/SqlCreateSnapshot.java:55]: I don't like the indirection much, I don't see a benefit to it. It technically isn't a validator exception, but parsing exc: the parser fails to create a SqlNode.
We do this style already, for example here: https://github.com/hazelcast/hazelcast-jet/blob/master/hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/SqlCreateJob.java#L117
The user will receive `HazelcastSqlException` with `code=PARSING` anyway.

Comment (gierlachg) [reply hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/SqlCreateSnapshot.java:55]: Exactly, if it's a parsing one maybe we should throw a _parsing_ exception, giving user position etc., `QueryException` does not seem to be fitted to that.

Comment (viliam-durina) [reply hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/SqlCreateSnapshot.java]: Got it, fixed. But the exception marks the whole command...

**Review by gierlachg — APPROVED** [review]

**Review by mtopolnik — APPROVED** [review]

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/3552/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/3552/jet-oss-pr-builder-3552.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] COMPILATION ERROR : 
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetSqlBackend.java:[169,33] cannot find symbol
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile (default-compile) on project hazelcast-jet-sql: Compilation failure: Compilation failure: 
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetSqlBackend.java:[169,33] cannot find symbol
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetSqlBackend.java:[184,50] cannot find symbol
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/target/generated-sources/java/com/hazelcast/jet/sql/impl/calcite/parser/JetSqlParser.java:[4616,41] incompatible types: org.apache.calcite.sql.SqlIdentifier cannot be converted to java.lang.String
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/target/generated-sources/java/com/hazelcast/jet/sql/impl/calcite/parser/JetSqlParser.java:[4629,57] incompatible types: org.apache.calcite.sql.SqlIdentifier cannot be converted to java.lang.String
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/target/generated-sources/java/com/hazelcast/jet/sql/impl/calcite/parser/JetSqlParser.java:[4630,129] incompatible types: org.apache.calcite.sql.SqlIdentifier cannot be converted to java.lang.String
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetSqlBackend.java:[169,33] cannot find symbol
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/JetSqlBackend.java:[184,50] cannot find symbol
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/target/generated-sources/java/com/hazelcast/jet/sql/impl/calcite/parser/JetSqlParser.java:[4616,41] incompatible types: org.apache.calcite.sql.SqlIdentifier cannot be converted to java.lang.String
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/target/generated-sources/java/com/hazelcast/jet/sql/impl/calcite/parser/JetSqlParser.java:[4629,57] incompatible types: org.apache.calcite.sql.SqlIdentifier cannot be converted to java.lang.String
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/target/generated-sources/java/com/hazelcast/jet/sql/impl/calcite/parser/JetSqlParser.java:[4630,129] incompatible types: org.apache.calcite.sql.SqlIdentifier cannot be converted to java.lang.String
--------------------------
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/3553/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/3553/jet-oss-pr-builder-3553.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] COMPILATION ERROR : 
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/target/generated-sources/java/com/hazelcast/jet/sql/impl/calcite/parser/JetSqlParser.java:[4616,41] incompatible types: org.apache.calcite.sql.SqlIdentifier cannot be converted to java.lang.String
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile (default-compile) on project hazelcast-jet-sql: Compilation failure: Compilation failure: 
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/target/generated-sources/java/com/hazelcast/jet/sql/impl/calcite/parser/JetSqlParser.java:[4616,41] incompatible types: org.apache.calcite.sql.SqlIdentifier cannot be converted to java.lang.String
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/target/generated-sources/java/com/hazelcast/jet/sql/impl/calcite/parser/JetSqlParser.java:[4629,57] incompatible types: org.apache.calcite.sql.SqlIdentifier cannot be converted to java.lang.String
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/target/generated-sources/java/com/hazelcast/jet/sql/impl/calcite/parser/JetSqlParser.java:[4630,129] incompatible types: org.apache.calcite.sql.SqlIdentifier cannot be converted to java.lang.String
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/target/generated-sources/java/com/hazelcast/jet/sql/impl/calcite/parser/JetSqlParser.java:[4616,41] incompatible types: org.apache.calcite.sql.SqlIdentifier cannot be converted to java.lang.String
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/target/generated-sources/java/com/hazelcast/jet/sql/impl/calcite/parser/JetSqlParser.java:[4629,57] incompatible types: org.apache.calcite.sql.SqlIdentifier cannot be converted to java.lang.String
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-sql/target/generated-sources/java/com/hazelcast/jet/sql/impl/calcite/parser/JetSqlParser.java:[4630,129] incompatible types: org.apache.calcite.sql.SqlIdentifier cannot be converted to java.lang.String
--------------------------
</details>
</pre>

**Pull request merged**